src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8EncodedCodePoint.cs

```diff
@@ -33,6 +33,27 @@ public Utf8EncodedCodePoint(char character) : this()
             }
         }
 
+        public Utf8EncodedCodePoint(UnicodeCodePoint codePoint) : this()
+        {
+            if (UnicodeCodePoint.IsSurrogate(codePoint))
+            {
+                throw new ArgumentOutOfRangeException("character", "Surrogate characters are not allowed");
+            }
+
+            unsafe
+            {
+                fixed (byte* encodedData = &_byte0)
+                {
+                    var buffer = new Span<byte>(encodedData, 4);
+                    if (!Utf8Encoder.TryEncodeCodePoint(codePoint, buffer, out _length))
```

Comment:
How fast do we need this ctor to be? If we need it to be fast, I think you should implement custom routine to convert UCP to each of the fields. I worry that creating span and then calling the TryEncodeCodePoint method will be too slow. But maybe you should measure.